feat(popover): add guidelines and code for popover (#245)
siemens/ix-docs: docs/components/popover/guide.md

---
doc-type: 'tab-item'
description: 'Usage guide for the ix-popover'
---
# Popover - Usage

> Use popovers when users need extra context without leaving the current task or losing sight of the trigger. We recommend popovers for short, contextual content that helps users decide, confirm or learn more in place.

Reviewer: "decide": You might reframe this a bit
1. In the dos section you recommend only 1 action
2. up till now, decisions are done in modal messages - do we now recommend both? (if so, I'd add a related-link to the message modal)

Reviewer 2: I will change "decide" to "act" to keep the guidance for important decisions in the modal message. Focus here should be that popovers provide help to those actions.

![Anatomy component](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7956-140&t=ekGYaTlPLbfy7TSp-4)

1. Header with popover title (optional)
2. Image (optional)
3. Content

> 4. Footer (optional)

Reviewer: I wasnt aware that we add the information about optional 🙊 we might need to add that to several components if needed

Reviewer 2: removed optional

5. Spike (optional)

> ## Options

Reviewer: Consider grouping the actions by slot, or add the slot name to the option (e.g. alignment wasn't clear at first glance that it's only about the footer)

Reviewer 2: sorted the options and added the slot name

> ## Options

Reviewer: header icon property is missing; also consider additional items slot and start slot inside the footer

Reviewer 2: added

- **Trigger:** Use a clear trigger that signals additional context, e.g. an info icon, status chip or secondary action

> - **Placement:** We typically use the default bottom placement because it feels connected to the trigger and works well in dense layouts

Reviewer: ```suggestion
- **Placement:** We typically use the default bottom placement because it follows the natural reading order.
```
Not sure how bottom placement works better than any other placement for dense layouts?

Reviewer 2: taken over suggestion

> - **Has spike:** Use the spike when users need a stronger visual connection between the popover and its trigger

Reviewer: Just as a consideration, here's the rule in carbon that might be better applicable: arrows are most helpful when the trigger lacks a strong selected state or is a compact icon trigger, and less necessary when the trigger already clearly reads as active.

Reviewer 2: good idea, adapted

> - **Trigger mode:** Prefer click for interactive content. Use hover only for lightweight, non-essential information

Reviewer: The hover-definition basically describes tooltips, isnt it?

Reviewer 2: yes, but also possible here. could consider deprecating the tooltip later.

- **Close on click outside:** Keep this enabled for temporary, dismissible information so users can return to the page quickly
- **Title:** Add a short title when users need to understand the topic before scanning the content

> - **Image:** Use images sparingly and only when they clarify the message faster than text alone

Reviewer: Consider be a bit more concrete, e.g.
titles: specific and scannable
images: never be decorative in a small transient overlay, no text - add alt-texts

Reviewer 2: added

> - **Footer actions:** Add footer actions only when users need a lightweight next step, e.g. dismissing a hint or opening more details

Reviewer: In other systems, stacked or numerous actions are usually treated as a warning sign, not just a spacing choice - consider defining how many actions are acceptable

Reviewer 2: I added the recommended limit to three to match our primary, secondary, tertiary buttons

- **Alignment:** Use horizontal footer actions by default. Switch to vertical only when labels are long or space is constrained

> ## Behavior in context

Reviewer: we typically use this section not for recommendations but how the component behaves in context of other elements (e.g. keep text concise should be placed on the title). As a result, what do you think about describing the focus-logic here?
* if there are interactive element inside: focus goes from trigger element to first interactive element, when closing back to trigger element
* if there are no interactive elements: focus stays on trigger

Reviewer 2: removed or the recommendation style and added the focus logic

- **Overflow:**
    - Keep text concise to avoid large floating panels that cover too much surrounding content
    - If content grows beyond a short explanation or a small set of actions, we recommend using a [modal](../modal) instead
    - Use images, metadata and actions only when they support the main message rather than competing with it
- **Placement:**
    - Popovers anchor to the trigger element, so placement should preserve that visual relationship
    - We usually place popovers below the trigger. When space is limited near the viewport edge, the placement adjusts automatically
- **Responsiveness:**
    - On smaller viewports, keep the content especially brief so the popover doesn’t obscure the surrounding interface
    - If users need to compare information, read long text or complete multiple actions on mobile, we recommend switching to a modal or dedicated page

## States

Popovers have two states: Closed and opened. When a popover is opened, it appears anchored to the trigger and is dismissible so users can return to the underlying context quickly.

> ## Dos and Don’ts

Reviewer: How about multiple popovers at once? Or nesting dropdowns (e.g. a select or date input)?

Reviewer 2: good idea for don'ts to avoid focus complexity. added it

- Do use popovers for contextual information, e.g. release highlights or guided hints
- Do use popovers when users need one lightweight action without leaving the current task
- Do use a short title and concise content so users can scan the message quickly

> - Do use tooltips for brief, non-interactive hints on hover or focus

Reviewer: tooltips?

Reviewer 2: removed

---
doc-type: 'tab-item'
description: 'Usage guide for the ix-popover'
---
# Popover - Usage

Use popovers when users need extra context without leaving the current task or losing sight of the trigger. We recommend popovers for short, contextual content that helps users act, confirm or learn more in place.

![Anatomy component](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7956-140&t=ekGYaTlPLbfy7TSp-4)

1. Header with popover title
2. Image
3. Content
4. Footer

> 5. Spike

Reviewer 3: is spike well enough known to not add any other info?

## Options

> - **Trigger:** Use a clear trigger that signals additional context, e.g. an info icon, status chip or secondary action

Reviewer 3: add full stops to all the sentences. Make sure you also add full stops to all the second sentences. We approach this with all or none

> - **Trigger mode:** Prefer click for interactive content. Use hover only for lightweight, non-essential information

Reviewer 3: "prefer" not used in command mood like this grammatically. I think you mean: "We recommend using click for interactive content." / "Use click for interactive content."

- **Placement:** We typically use the default bottom placement because it follows the natural reading order.
- **Close on click outside:** Keep this enabled for temporary, dismissible information so users can return to the page quickly
- **Has spike:** Use the spike when the trigger has a weak selected state or is a compact icon trigger. If the trigger already reads clearly as active, the spike is often unnecessary
- **Header:** Use the header to establish context before users scan the body content
    - **Title:** Add a short title when users need to understand the topic before scanning the content. Keep titles specific and scannable
    - **Header icon:** An icon can be placed in the header to support quick recognition of the message context. Use it only when it adds meaning and avoid purely decorative icons
    - **Additional header items slot:** Add compact supporting items beside the title, e.g. a status pill, only when they help users interpret the message faster

> - **Image:** Use images sparingly and only when they clarify the message faster than text alone. In a small transient overlay, images should never be decorative, should not contain text, and should include meaningful alt text

Reviewer 3: ```suggestion
- **Image:** Use images sparingly and only when they clarify the message faster than text alone. In a small transient overlay, images should never be decorative, contain text, and always include meaningful alt text.
```

- **Placement:**
    - Popovers are anchored to the trigger element and positioned to keep that relationship clear
    - When there is not enough space near the viewport edge, the position can flip to keep the panel visible
- **Responsiveness:**
    - On smaller viewports, a popover can overlap a larger share of the interface around the trigger
    - If users need to compare information, read longer content, or complete multiple actions, a [modal](../modal) or dedicated page supports the flow better

## States

Popovers have two states: Closed and opened. When a popover is opened, it appears anchored to the trigger and is dismissible so users can return to the underlying context quickly.

## Dos and Don’ts

- Do use popovers for contextual information, e.g. release highlights or guided hints
- Do use popovers when users need one lightweight action without leaving the current task
- Do use a short title and concise content so users can scan the message quickly
- Don’t use popovers for essential information that must be read before proceeding, use [modals](../modal) instead
- Don’t use popovers for selecting from a list of actions, use [dropdowns](../dropdown) instead

> - Don’t keep multiple popovers open at the same time, close the current one before opening another

Reviewer 3: ```suggestion
- Don’t allow multiple popovers to be open at the same time, close one before opening another
```

> - Don’t allow multiple popovers to be open at the same time, close the current one before opening another

Reviewer 4: Are nesting popovers also affected by this discussion?

Reviewer: This don't is a separate topic; we had an actual _don't_ for nesting popovers but removed it to keep the list neat and tidy.

- Do use popovers for contextual information, e.g. release highlights or guided hints
- Do use popovers when users need one lightweight action without leaving the current task
- Do use a short title and concise content so users can scan the message quickly
- Don’t use popovers for essential information that must be read before proceeding, use [modals](../modal) instead
- Don’t use popovers for selecting from a list of actions, use [dropdowns](../dropdown) instead
- Don’t keep multiple popovers open at the same time, close the current one before opening another
- Don’t open a [dropdown](../dropdown) inside a popover when users need many choices, use a modal or dedicated page instead

> - Don’t use popovers for long explanations or multi-step tasks, use a [modal](../modal) or dedicated page instead

Reviewer: we try to have more do's than don'ts, or have them at least even. Is that any way to remove 3 donts?

Reviewer 2: yes, done
